s4/clarity/configuration/process_type.py

```diff
@@ -1,46 +1,182 @@
+
+
+class ProcessOutput(WrappedXml):
+    """
+    Represents a process-output in Clarity
+
+    https://www.genologics.com/files/permanent/API/latest/data_ptp.html#process-output
+    """
+
+    artifact_type = subnode_property('artifact-type')
+    display_name = subnode_property('display-name')
+    output_generation_type = subnode_property('output-generation-type')
+    number_of_outputs = subnode_property('number-of-outputs', typename=types.NUMERIC)
+    output_name = subnode_property('output-name')
+    assign_working_flag = subnode_property('assign-working-flag', typename=types.BOOLEAN)
+    field_definitions = subnode_links(Udf, 'field-definition')
+
+
+class Parameter(WrappedXml):
+    """
+    Represents a parameter in Clarity
+
+    https://www.genologics.com/files/permanent/API/latest/data_ptp.html#parameter
+    """
+
+    name = attribute_property('name')
+    command_string = subnode_property('string')
+    run_program_per_event = subnode_property('run-program-per-event', typename=types.BOOLEAN)
+    channel = subnode_property('channel')
+    invocation_type = subnode_property('invocation-type')
+    # ToDo: add file list
+
+
+class QueueField(Field):
+    """
+    Represents a queuefield in Clarity
+
+    https://www.genologics.com/files/permanent/API/latest/data_ptp.html#queuefield
+    """
+
+    detail = attribute_property('detail')
+
+
+class IceBucketField(Field):
+    """
+    Represents a icebucketfield in Clarity
+
+    https://www.genologics.com/files/permanent/API/latest/data_ptp.html#icebucketfield
+    """
+
+    detail = attribute_property('detail')
+
 
-    inputs = subnode_property_list_of_dicts('process-input', as_attributes=["name"])
+class ProcessTypeAttribute(WrappedXml):
+    """
+    Represents a process-type-attribute in Clarity
 
-    outputs = subnode_property_list_of_dicts('process-output', as_attributes=["name", "uri"])
+    https://www.genologics.com/files/permanent/API/latest/data_ptp.html#process-type-attribute
+    """
 
-    parameters = subnode_property_list_of_dicts('parameter', as_attributes=["name"])
+    name = attribute_property('name')
+    value = subnode_property(".")
+
+
+class ProcessStepProperty(WrappedXml):
+    """
+    Represents a step-property in Clarity
+
+    https://www.genologics.com/files/permanent/API/latest/data_ptp.html#step-property
+    """
+
+    name = attribute_property('name')
+    value = attribute_property('value')
+
+
+class ProcessEppTrigger(WrappedXml):
+    """
+    Represents an epp-trigger in Clarity
+
+    https://www.genologics.com/files/permanent/API/latest/data_ptp.html#epp-trigger
+    """
+
+    name = attribute_property('name')
+    status = attribute_property('status')
+    point = attribute_property('point')
+    type = attribute_property('type')
+
+class ProcessPermittedInstrumentType(WrappedXml):
+    """
+    Represents an Instrument Type definition in the Process type.
+
+    This field is a string.
```

Comment:
This is a really good example of my general review comment - what do you get out of this being a class?

Reply:
I am unsure of how we would expose the Instrument fields otherwise. It is a variable length list of subnodes, and while they all only have a single string inside them I can not pass `str` as the type to the `subnode_element_list` because it does not know how to take the ETree.Element and turn it into the value we want. 

We also can not use the more generic dictionary objects to parse it because there is no key associated with any of the values. I would prefer that we exposed this as a list of strings but I could not find a way to do that with out creating a new property accessor class.

In a more general response to what we get out of it being a class, we gain the use of the more modern accessors, such as `subnode_element_list`, `subnode_element` etc. which provide a more discoverable interface for the user.